src/sidebar/views/reports.ts

```diff
@@ -0,0 +1,249 @@
+import { basename } from 'path';
+import {
+    Command,
+    Event,
+    EventEmitter,
+    ExtensionContext,
+    TreeDataProvider,
+    TreeItem,
+    TreeItemCollapsibleState,
+    TreeView,
+    Uri,
+    commands,
+    window
+} from 'vscode';
+import { ExtensionApi } from '../../backend/api';
+import { AnalysisPathEvent, AnalysisPathKind, DiagnosticEntry } from '../../backend/types';
+
+export interface IssueMetadata {
+    entryIndex?: number;
+    reprStep?: number;
+    reprHasChildren?: boolean;
+    description?: string;
+    command?: Command;
+}
+
+export class ReportsView implements TreeDataProvider<IssueMetadata> {
+    protected currentFile?: Uri;
+    protected currentEntryList?: DiagnosticEntry[];
+
+    protected tree?: TreeView<IssueMetadata>;
+
+    constructor(ctx: ExtensionContext) {
+        ctx.subscriptions.push(this._onDidChangeTreeData = new EventEmitter());
+        window.onDidChangeActiveTextEditor(this.refreshBugList, this, ctx.subscriptions);
+        ExtensionApi.diagnostics.diagnosticsUpdated(this.refreshBugList, this, ctx.subscriptions);
+
+        ctx.subscriptions.push(this.tree = window.createTreeView(
+            'codechecker.views.reports',
+            { treeDataProvider: this }
+        ));
+
+        this.init();
+    }
+
+    protected init() {
+        this.currentFile = window.activeTextEditor?.document.uri;
+    }
+
+    private _onDidChangeTreeData: EventEmitter<void>;
+    public get onDidChangeTreeData(): Event<void> {
+        return this._onDidChangeTreeData.event;
+    }
+
+    refreshBugList() {
+        // Hide the report list when there's no metadata
+        if (!ExtensionApi.metadata.metadata) {
+            commands.executeCommand('setContext', 'codechecker.sidebar.showReports', false);
+        }
+
+        this.currentFile = window.activeTextEditor?.document.uri;
+
+        // Clear tree on file close
+        if (this.currentFile === undefined) {
+            this.currentEntryList = [];
+            this._onDidChangeTreeData.fire();
+            return;
+        }
+
+        this.currentEntryList = ExtensionApi.diagnostics.getFileDiagnostics(this.currentFile);
+        this._onDidChangeTreeData.fire();
+
+        commands.executeCommand('setContext', 'codechecker.sidebar.showReports', true);
+    }
+
+    getChildren(element?: IssueMetadata): IssueMetadata[] | undefined {
+        const makeArray = <T>(length: number, func: (idx: number) => T): T[] => {
```

Comment:
This variable / function is not used so it can be removed. To avoid such problems please enable this rule: https://eslint.org/docs/rules/no-unused-vars